client/context.go

```diff
@@ -89,6 +91,100 @@ func NewContextWithInput(input io.Reader) Context {
 	return NewContextWithInputAndFrom(input, viper.GetString(flags.FlagFrom))
 }
 
+// Context function With FlagSet functions will deprecate old new context that depend on
+// viper.
+// TODO after viper dependency is removed this functions should be the only one that are called.
+
+// NewContextWithFsAndInput return an initialized context from flagset and input.
+func NewContextWithFsAndInput(fs *pflag.FlagSet, input io.Reader) (Context, error) {
```

Comment:
It was in order to remove every viper reference, we can leave it like this ultil remove it. I just wanted to be sure in all the complete flow there was no hidden viper calls.

I needed it because gentx command used this and inside was full of viper calls. We can always remove it.

https://github.com/cosmos/cosmos-sdk/blob/747965f809e22f2a5c5696b47b7a94b3084815f4/x/genutil/client/cli/gentx.go#L136

Reply:
Can we avoid this and just create a `Context` literal and use `cmd.Flags()` to populate it?

Reply:
Let me try

Reply:
Thanks @alexanderbez now I understand all those Init* and New* funcs. Already removed! Much more clear now!